<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The QFeatures class • QFeaturesScpWorkshop2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The QFeatures class">
<meta property="og:description" content="QFeaturesScpWorkshop2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">QFeaturesScpWorkshop2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v01-QFeatures.html">The QFeatures class</a>
    </li>
    <li>
      <a href="../articles/v02-scp.html">Using scp for single-cell proteomics analyses</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lgatto/QFeaturesScpWorkshop2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="v01-QFeatures_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>The QFeatures class</h1>
                        <h4 class="author">Laurent Gatto and Christophe Vanderaa</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lgatto/QFeaturesScpWorkshop2021/blob/master/vignettes/v01-QFeatures.Rmd"><code>vignettes/v01-QFeatures.Rmd</code></a></small>
      <div class="hidden name"><code>v01-QFeatures.Rmd</code></div>

    </div>

    
    
<p><strong>Last modified:</strong> NA<br><strong>Compiled</strong>: Wed Jun 30 21:48:17 2021</p>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><code>QFeatures</code> is a package for the manipulation of mass spectrometry (MS)-based quantitative proteomics data. The quantitative data is generally obtained after running a preprocessing algorithm on MS spectra. This algorithm matches the spectra to a peptide database to infer the peptide sequence from which the spectra originated. Spectra for which a corresponding peptide could be found are called peptide to spectrum matches (PSMs). The intensities recorded by the MS are then used to quantify the PSMs. The quantified PSMs are used as input by <code>QFeatures</code> and undergo various data wrangling steps to reconstruct the peptide data and then the protein data, usually of interest for elucidating the biological question at hand.</p>
<p>MS-based quantification data can be represented as a matrix of quantitative values for features (PSMs, peptides, proteins) arranged along the rows, measured for a set of samples, arranged along the columns. The matrix format is a common representation for any quantitative data set. We will be using the <code>SummarizedExperiment</code> <span class="citation">(Morgan et al. 2020)</span> class:</p>
<div class="figure">
<img src="figs/SE.png" alt="Schematic representation of the anatomy of a `SummarizedExperiment` object. (Figure taken from the `SummarizedExperiment` package vignette.)" width="353"><p class="caption">
Schematic representation of the anatomy of a <code>SummarizedExperiment</code> object. (Figure taken from the <code>SummarizedExperiment</code> package vignette.)
</p>
</div>
<ul>
<li>The sample (columns) metadata can be access with the <code>colData()</code> function.</li>
<li>The features (rows) metadata can be access with the <code>rowData()</code> column.</li>
<li>If the features represent ranges along genomic coordinates, these can be accessed with <code>rowRanges()</code>
</li>
<li>Additional metadata describing the overall experiment can be accessed with <code>metadata()</code>.</li>
<li>The quantiative data can be accessed with <code>assay()</code>.</li>
<li>
<code>assays()</code> returns a list of matrix-like assays.</li>
</ul>
</div>
<div id="qfeatures" class="section level2">
<h2 class="hasAnchor">
<a href="#qfeatures" class="anchor"></a>QFeatures</h2>
<p>While mass spectrometers acquire data for spectra/peptides, the biological entity of interest are the protein. As part of the data processing, we are thus required to <strong>aggregate</strong> low-level quantitative features into higher level data.</p>
<div class="figure">
<img src="v01-QFeatures_files/figure-html/featuresplot-1.png" alt="Conceptual representation of a `QFeatures` object and the aggregative relation between different assays." width="700"><p class="caption">
Conceptual representation of a <code>QFeatures</code> object and the aggregative relation between different assays.
</p>
</div>
<p>We are going to start to familiarise ourselves with the <code>QFeatures</code> class implemented in the <a href="https://rformassspectrometry.github.io/QFeatures/articles/QFeatures.html"><code>QFeatures</code></a> package. The class is derived from the Bioconductor <code>MultiAssayExperiment</code> <span class="citation">(Ramos et al. 2017)</span> class. Let’s start by loading the <code>QFeatures</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/RforMassSpectrometry/QFeatures">"QFeatures"</a></span><span class="op">)</span></code></pre></div>
<p>Next, we load the <code>feat1</code> test data, which is composed of single <em>assay</em> of class <code>SummarizedExperiment</code> composed of 10 rows and 2 columns.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">feat1</span><span class="op">)</span>
<span class="va">feat1</span>
<span class="co">## An instance of class QFeatures containing 1 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 10 rows and 2 columns</span></code></pre></div>
<p>Let’s perform some simple operations to familiarise ourselves with the <code>QFeatures</code> class:</p>
<ul>
<li>Extract the sample metadata using the <code>colData()</code> accessor (like you have previously done with <code>SummarizedExperiment</code> objects).</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">colData</span><span class="op">(</span><span class="va">feat1</span><span class="op">)</span>
<span class="co">## DataFrame with 2 rows and 1 column</span>
<span class="co">##        Group</span>
<span class="co">##    &lt;integer&gt;</span>
<span class="co">## S1         1</span>
<span class="co">## S2         2</span></code></pre></div>
<ul>
<li>Extract the first (and only) assay composing this <code>QFeaures</code> data using the <code><a href="https://rdrr.io/r/base/Extract.html">[[</a></code> operator (as you have done to extract elements of a list) by using the assay’s index or name.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">feat1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 10 2 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(10): PSM1 PSM2 ... PSM9 PSM10</span>
<span class="co">## rowData names(5): Sequence Protein Var location pval</span>
<span class="co">## colnames(2): S1 S2</span>
<span class="co">## colData names(0):</span>
<span class="va">feat1</span><span class="op">[[</span><span class="st">"psms"</span><span class="op">]</span><span class="op">]</span>
<span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 10 2 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(10): PSM1 PSM2 ... PSM9 PSM10</span>
<span class="co">## rowData names(5): Sequence Protein Var location pval</span>
<span class="co">## colnames(2): S1 S2</span>
<span class="co">## colData names(0):</span></code></pre></div>
<ul>
<li>Extract the <code>psms</code> assay’s row data and quantitative values.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">##       S1 S2</span>
<span class="co">## PSM1   1 11</span>
<span class="co">## PSM2   2 12</span>
<span class="co">## PSM3   3 13</span>
<span class="co">## PSM4   4 14</span>
<span class="co">## PSM5   5 15</span>
<span class="co">## PSM6   6 16</span>
<span class="co">## PSM7   7 17</span>
<span class="co">## PSM8   8 18</span>
<span class="co">## PSM9   9 19</span>
<span class="co">## PSM10 10 20</span>
<span class="fu">rowData</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">## DataFrame with 10 rows and 5 columns</span>
<span class="co">##            Sequence     Protein       Var      location      pval</span>
<span class="co">##         &lt;character&gt; &lt;character&gt; &lt;integer&gt;   &lt;character&gt; &lt;numeric&gt;</span>
<span class="co">## PSM1       SYGFNAAR       ProtA         1 Mitochondr...     0.084</span>
<span class="co">## PSM2       SYGFNAAR       ProtA         2 Mitochondr...     0.077</span>
<span class="co">## PSM3       SYGFNAAR       ProtA         3 Mitochondr...     0.063</span>
<span class="co">## PSM4       ELGNDAYK       ProtA         4 Mitochondr...     0.073</span>
<span class="co">## PSM5       ELGNDAYK       ProtA         5 Mitochondr...     0.012</span>
<span class="co">## PSM6       ELGNDAYK       ProtA         6 Mitochondr...     0.011</span>
<span class="co">## PSM7  IAEESNFPFI...       ProtB         7       unknown     0.075</span>
<span class="co">## PSM8  IAEESNFPFI...       ProtB         8       unknown     0.038</span>
<span class="co">## PSM9  IAEESNFPFI...       ProtB         9       unknown     0.028</span>
<span class="co">## PSM10 IAEESNFPFI...       ProtB        10       unknown     0.097</span></code></pre></div>
</div>
<div id="feature-aggregation" class="section level2">
<h2 class="hasAnchor">
<a href="#feature-aggregation" class="anchor"></a>Feature aggregation</h2>
<p>The central functionality of the <code>QFeatures</code> infrastructure is the aggregation of features into higher-level features while retaining the link between the different levels. This can be done with the <a href="https://rformassspectrometry.github.io/QFeatures/reference/QFeatures-aggregate.html"><code>aggregateFeatures()</code> function</a>.</p>
<p>The call below will</p>
<ul>
<li>operate on the <code>psms</code> assay of the <code>feat1</code> objects;</li>
<li>aggregate the rows the assay following the grouping defined in the <code>peptides</code> row data variables;</li>
<li>perform aggregation using the <code><a href="https://rdrr.io/r/base/colSums.html">colMeans()</a></code> function;</li>
<li>create a new assay named <code>peptides</code> and add it to the <code>feat1</code> object.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">feat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">feat1</span>, i <span class="op">=</span> <span class="st">"psms"</span>,
                           fcol <span class="op">=</span> <span class="st">"Sequence"</span>,
                           name <span class="op">=</span> <span class="st">"peptides"</span>,
                           fun <span class="op">=</span> <span class="va">colMeans</span><span class="op">)</span>
<span class="va">feat1</span>
<span class="co">## An instance of class QFeatures containing 2 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 10 rows and 2 columns </span>
<span class="co">##  [2] peptides: SummarizedExperiment with 3 rows and 2 columns</span></code></pre></div>
<ul>
<li>Let’s convince yourself that we understand the effect of feature aggregation and repeat the calculations manually and check the content of the new assay’s row data.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## SYGFNAAR</span>
<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">## S1 S2 </span>
<span class="co">##  2 12</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="st">"SYGFNAAR"</span>, <span class="op">]</span>
<span class="co">## S1 S2 </span>
<span class="co">##  2 12</span>

<span class="co">## ELGNDAYK</span>
<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">## S1 S2 </span>
<span class="co">##  5 15</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="st">"ELGNDAYK"</span>, <span class="op">]</span>
<span class="co">## S1 S2 </span>
<span class="co">##  5 15</span>

<span class="co">## IAEESNFPFIK</span>
<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">7</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span><span class="op">)</span>
<span class="co">##   S1   S2 </span>
<span class="co">##  8.5 18.5</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="st">"IAEESNFPFIK"</span>, <span class="op">]</span>
<span class="co">##   S1   S2 </span>
<span class="co">##  8.5 18.5</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">## DataFrame with 3 rows and 4 columns</span>
<span class="co">##                  Sequence     Protein      location        .n</span>
<span class="co">##               &lt;character&gt; &lt;character&gt;   &lt;character&gt; &lt;integer&gt;</span>
<span class="co">## ELGNDAYK         ELGNDAYK       ProtA Mitochondr...         3</span>
<span class="co">## IAEESNFPFIK IAEESNFPFI...       ProtB       unknown         4</span>
<span class="co">## SYGFNAAR         SYGFNAAR       ProtA Mitochondr...         3</span></code></pre></div>
<p>We can now aggregate the peptide-level data into a new protein-level assay using the <code>colMedians()</code> aggregation function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">feat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-aggregate.html">aggregateFeatures</a></span><span class="op">(</span><span class="va">feat1</span>, i <span class="op">=</span> <span class="st">"peptides"</span>,
                           fcol <span class="op">=</span> <span class="st">"Protein"</span>,
                           name <span class="op">=</span> <span class="st">"proteins"</span>,
                           fun <span class="op">=</span> <span class="va">colMedians</span><span class="op">)</span>
<span class="va">feat1</span>
<span class="co">## An instance of class QFeatures containing 3 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 10 rows and 2 columns </span>
<span class="co">##  [2] peptides: SummarizedExperiment with 3 rows and 2 columns </span>
<span class="co">##  [3] proteins: SummarizedExperiment with 2 rows and 2 columns</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">feat1</span><span class="op">[[</span><span class="st">"proteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">##        S1   S2</span>
<span class="co">## ProtA 3.5 13.5</span>
<span class="co">## ProtB 8.5 18.5</span></code></pre></div>
</div>
<div id="subsetting-and-filtering" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting-and-filtering" class="anchor"></a>Subsetting and filtering</h2>
<p>The link between the assays becomes apparent when we now subset the assays for protein A as shown below or using the <code><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-subsetBy.html">subsetByFeature()</a></code> function. This creates a new instance of class <code>QFeatures</code> containing assays with the expression data for protein, its peptides and their PSMs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">feat1</span><span class="op">[</span><span class="st">"ProtA"</span>, , <span class="op">]</span>
<span class="co">## An instance of class QFeatures containing 3 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 6 rows and 2 columns </span>
<span class="co">##  [2] peptides: SummarizedExperiment with 2 rows and 2 columns </span>
<span class="co">##  [3] proteins: SummarizedExperiment with 1 rows and 2 columns</span></code></pre></div>
<p>The <code><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-filtering.html">filterFeatures()</a></code> function can be used to filter rows the assays composing a <code>QFeatures</code> object using the row data variables. We can for example retain rows that have a <code>pval</code> &lt; 0.05, which would only keep rows in the <code>psms</code> assay because the <code>pval</code> is only relevant for that assay.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">feat1</span>, <span class="op">~</span> <span class="va">pval</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="co">## An instance of class QFeatures containing 3 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 4 rows and 2 columns </span>
<span class="co">##  [2] peptides: SummarizedExperiment with 0 rows and 2 columns </span>
<span class="co">##  [3] proteins: SummarizedExperiment with 0 rows and 2 columns</span></code></pre></div>
<p>On the other hand, if we filter assay rows for those that localise to the mitochondrion, we retain the relevant protein, peptides and PSMs.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">feat1</span>, <span class="op">~</span> <span class="va">location</span> <span class="op">==</span> <span class="st">"Mitochondrion"</span><span class="op">)</span>
<span class="co">## An instance of class QFeatures containing 3 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 6 rows and 2 columns </span>
<span class="co">##  [2] peptides: SummarizedExperiment with 2 rows and 2 columns </span>
<span class="co">##  [3] proteins: SummarizedExperiment with 1 rows and 2 columns</span></code></pre></div>
<p>As an exercise, let’s filter rows that do not localise to the mitochondrion.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/QFeatures/man/QFeatures-filtering.html">filterFeatures</a></span><span class="op">(</span><span class="va">feat1</span>, <span class="op">~</span> <span class="va">location</span> <span class="op">!=</span> <span class="st">"Mitochondrion"</span><span class="op">)</span>
<span class="co">## An instance of class QFeatures containing 3 assays:</span>
<span class="co">##  [1] psms: SummarizedExperiment with 4 rows and 2 columns </span>
<span class="co">##  [2] peptides: SummarizedExperiment with 1 rows and 2 columns </span>
<span class="co">##  [3] proteins: SummarizedExperiment with 1 rows and 2 columns</span></code></pre></div>
<p>You can refer to the <a href="https://rformassspectrometry.github.io/QFeatures/articles/QFeatures.html"><em>Quantitative features for mass spectrometry data</em></a> vignette and the <code>QFeature</code> <a href="https://rformassspectrometry.github.io/QFeatures/reference/QFeatures-class.html">manual page</a> for more details about the class.</p>
</div>
<div id="session-information" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.1.0 (2021-05-18)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.2 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] QFeatures_1.3.3             MultiAssayExperiment_1.19.1</span>
<span class="co">##  [3] SummarizedExperiment_1.23.1 Biobase_2.53.0             </span>
<span class="co">##  [5] GenomicRanges_1.45.0        GenomeInfoDb_1.29.2        </span>
<span class="co">##  [7] IRanges_2.27.0              S4Vectors_0.31.0           </span>
<span class="co">##  [9] BiocGenerics_0.39.1         MatrixGenerics_1.5.0       </span>
<span class="co">## [11] matrixStats_0.59.0          BiocStyle_2.21.3           </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] xfun_0.24               bslib_0.2.5.1           lattice_0.20-44        </span>
<span class="co">##  [4] htmltools_0.5.1.1       yaml_2.2.1              rlang_0.4.11           </span>
<span class="co">##  [7] pkgdown_1.6.1           jquerylib_0.1.4         GenomeInfoDbData_1.2.6 </span>
<span class="co">## [10] stringr_1.4.0           zlibbioc_1.39.0         ProtGenerics_1.25.0    </span>
<span class="co">## [13] ragg_1.1.3              memoise_2.0.0           evaluate_0.14          </span>
<span class="co">## [16] knitr_1.33              fastmap_1.1.0           highr_0.9              </span>
<span class="co">## [19] BiocManager_1.30.16     cachem_1.0.5            DelayedArray_0.19.1    </span>
<span class="co">## [22] desc_1.3.0              jsonlite_1.7.2          XVector_0.33.0         </span>
<span class="co">## [25] systemfonts_1.0.2       fs_1.5.0                MsCoreUtils_1.5.0      </span>
<span class="co">## [28] textshaping_0.3.5       png_0.1-7               digest_0.6.27          </span>
<span class="co">## [31] stringi_1.6.2           rprojroot_2.0.2         grid_4.1.0             </span>
<span class="co">## [34] clue_0.3-59             tools_4.1.0             bitops_1.0-7           </span>
<span class="co">## [37] magrittr_2.0.1          sass_0.4.0              RCurl_1.98-1.3         </span>
<span class="co">## [40] lazyeval_0.2.2          cluster_2.1.2           crayon_1.4.1           </span>
<span class="co">## [43] MASS_7.3-54             Matrix_1.3-4            rmarkdown_2.9          </span>
<span class="co">## [46] AnnotationFilter_1.17.0 R6_2.5.0                compiler_4.1.0</span></code></pre></div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-SE" class="csl-entry">
Morgan, Martin, Valerie Obenchain, Jim Hester, and Hervé Pagès. 2020. <em>SummarizedExperiment: SummarizedExperiment Container</em>. <a href="https://bioconductor.org/packages/SummarizedExperiment">https://bioconductor.org/packages/SummarizedExperiment</a>.
</div>
<div id="ref-MAE" class="csl-entry">
Ramos, Marcel, Lucas Schiffer, Angela Re, Rimsha Azhar, Azfar Basunia, Carmen Rodriguez Cabrera, Tiffany Chan, et al. 2017. <span>“Software for the Integration of Multi-Omics Experiments in Bioconductor.”</span> <em>Cancer Research</em> 77(21); e39-42.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Laurent Gatto, Christophe Vanderaa.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
